packages/react-storage/src/components/StorageBrowser/Views/Controls/Navigate.tsx

```diff
@@ -93,23 +88,53 @@ export interface NavigateControl<
 }
 
 export const NavigateControl: NavigateControl = (_props) => {
+  const [state, handleUpdateState] = useControl({ type: 'NAVIGATE' });
+
   return (
     <NavigateContainer>
-      <NavigateItem
-        item={{
-          name: 'Home',
-          permissionType: 'READ',
-          locationType: 'OBJECT',
-        }}
-      />
-      <NavigateItem
-        item={{
-          name: 'Location01',
-          permissionType: 'READ',
-          locationType: 'OBJECT',
-        }}
-        current
-      />
+      <NavigateItemContainer>
+        <NavigateButton
+          onClick={() => handleUpdateState({ type: 'DESELECT_LOCATION' })}
+        >
+          Home
+        </NavigateButton>
+        {state.location ? <Separator /> : null}
+      </NavigateItemContainer>
+      {state.location.current && (
+        <NavigateItemContainer>
+          <NavigateButton
+            onClick={() => {
+              if (state?.location.current) {
+                handleUpdateState({
+                  type: 'SELECT_LOCATION',
+                  location: state.location.current,
+                });
+              }
+            }}
+          >
+            {state.location.current.bucket}
+          </NavigateButton>
+          {state.history.list && state.history.list.length > 0 ? (
+            <Separator />
+          ) : null}
+        </NavigateItemContainer>
+      )}
```

Comment:
Defining the "Home" button and the "Bucket" manually here since there is only one instance of each. I did this so that `RenderItemProps` could have the `item` property only be a string and not a mix of types

Reply:
Need to have deterministic behavior here for flexibility